nixos/modules/services/networking/twingate.nix

```
@@ -0,0 +1,83 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.twingate;
+
+in {
+
+  options.services.twingate = {
+    enable = mkEnableOption "Twingate Client daemon";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.twingate;
+      defaultText = "pkgs.twingate";
+      description = "The package to use for Twingate";
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    networking.firewall.checkReversePath = lib.mkDefault false;
+
+    environment.systemPackages = [ cfg.package ]; # for the CLI
+    systemd.packages = [ cfg.package ];
+
+    system.activationScripts.twingate = stringAfter ["etc"] ''
+      mkdir -p '/etc/twingate'
+
+      cp -r -n ${cfg.package}/etc/twingate/. /etc/twingate/
+    '';
```

You don't need most of the files included in `${cfg.package}/etc/twingate`. And even if you did, you wouldn't want to copy over them every time.

```suggestion
    environment.etc."twingate/config.json".source = (pkgs.formats.json {}).generate "config.json" cfg.settings;
```

---

1) Regarding copying every time: it's done on purpose
2) Regarding config.json: it isn't basically a config, we'd like to fill through nix files. Maybe in future, but for now it's more a static data, which shouldn't be changed a lot

---

Are the files under `/etc/twingate` meant to be mutable or immutable? Is the directory `/etc/twingate` meant to be mutable or immutable? I'm having a bit of a hard time understanding the intended usage of this directory for your module. The `twingate` software merely reads from these files and doesn't write to them, correct?

---

@aanderse `/etc/twingate` should be mutable. We save user preferences there

---

@tonyshkurenko sorry this PR fell of my radar. So `twingate` manages its own configuration allowing you to modify it through its UI or something like that? I understand now... Thanks.